src/components/Table/GridTable.tsx

```diff
@@ -232,11 +233,21 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
   const { tableState } = api;
 
   tableState.onRowSelect = onRowSelect;
-  tableState.setRows(rows);
```

Comment:
I didn't really mean to tackle this in this PR, but after some of the other refactorings, on a whim I moved this `setRows` into a `useEffect` and it just worked/all tests passed. Which is great b/c this will avoid the React "render cannot set state in another component" errors.